### PR TITLE
feat: add error reporting tool using sentry

### PR DESCRIPTION
**Pull request by luizstacio**
This PR;
- Improves error handling for CRX
- Improve error handling UI to be less dependent on the application UI
- Add checking on database to report also CRX errors / crashes
- Fix the UI of the error handling screen related to the top items
- Change error reporting to use Sentry instead of email

How to test;
- Install the wallet extension;
- Open the service worker console, via the extension manager page link service worker.
- Execute the following command to dispatch a error `self.dispatchEvent(new ErrorEvent('error'));`
- Open the wallet, it should should the error reporting screen.

### CODE REVIEW

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

File: packages/app/.env.production [hunk 1]
```diff
@@ -3,5 +3,5 @@ VITE_FUEL_PROVIDER_URL=https://beta-3.fuel.network/graphql
 VITE_FUEL_FAUCET_URL=https://faucet-beta-3.fuel.network/
 VITE_MNEMONIC_WORDS=12
 VITE_FAUCET_RECAPTCHA_KEY=6Ld3cEwfAAAAAMd4QTs7aO85LyKGdgj0bFsdBfre
-VITE_ERROR_REPORT_EMAIL=contact@fuel.sh
 VITE_AUTO_LOCK_IN_MINUTES=10
+VITE_SENTRY_DNS=https://f9a5b923067f4f789fb40ed7eccd6486@o4505280621707264.ingest.sentry.io/4505280648577024
```
Comment (eswarasai) [inline packages/app/.env.production:7]:
```suggestion
VITE_SENTRY_DSN=https://f9a5b923067f4f789fb40ed7eccd6486@o4505280621707264.ingest.sentry.io/4505280648577024
```

File: packages/app/src/config.ts [hunk 2]
```diff
@@ -10,8 +10,8 @@ export const {
   VITE_DATABASE_VERSION,
   VITE_CRX_NAME,
   VITE_CRX,
-  VITE_ERROR_REPORT_EMAIL,
   VITE_AUTO_LOCK_IN_MINUTES,
+  VITE_SENTRY_DNS,
```
Comment (eswarasai) [inline packages/app/src/config.ts:14]:
```suggestion
  VITE_SENTRY_DSN,
```

File: packages/app/src/systems/Error/services/ReportErrorService.tsx [hunk 3]
```diff
@@ -1,40 +1,34 @@
 import type { FuelWalletError } from '@fuel-wallet/types';
+import * as Sentry from '@sentry/browser';
 
-import { parseFuelError, parseErrorEmail } from '../utils';
+import { parseFuelError } from '../utils';
 
-import { REPORT_ERROR_EMAIL } from '~/config';
-import { db } from '~/systems/Core/utils';
+import { VITE_SENTRY_DNS } from '~/config';
```
Comment (eswarasai) [inline packages/app/src/systems/Error/services/ReportErrorService.tsx:6]:
```suggestion
import { VITE_SENTRY_DSN } from '~/config';
```

File: packages/app/src/systems/Error/services/ReportErrorService.tsx [hunk 4]
```diff
@@ -1,40 +1,34 @@
...
   static async reportErrors() {
     const errors = await this.getErrors();
-    // send error as an email to the team
-    const errorMailBody = parseErrorEmail(errors);
-    window?.open(
-      `mailto:${REPORT_ERROR_EMAIL}?subject=Fuel Wallet Error Report&body=${errorMailBody}`,
-      '_blank'
-    );
+    Sentry.init({
+      dsn: VITE_SENTRY_DNS,
```
Comment (eswarasai) [inline packages/app/src/systems/Error/services/ReportErrorService.tsx:13]:
```suggestion
      dsn: VITE_SENTRY_DSN,
```

**Review by eswarasai — APPROVED** [review]
LGTM 👍

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
small change

**Review by eswarasai — APPROVED** [review]
LGTM 👍

**Review by LuizAsFight — APPROVED** [review]

**Pull request merged**